test/python/test_operation.py

```diff
@@ -2745,5 +2745,58 @@ def test_prelu_broadcast_cpu(self):
             np.testing.assert_array_almost_equal(tensor.to_numpy(tensor.from_raw_tensor(dx0)), grad0, decimal=5)
             np.testing.assert_array_almost_equal(tensor.to_numpy(tensor.from_raw_tensor(dx1)), grad1, decimal=5)
 
+    def test_gemm(self):
+        devs = [cpu_dev, gpu_dev]
```

Comment:
what if the machine does not have gpu?

Reply:
> what if the machine does not have gpu?

Yes, that's true.
Last time I run the test_operation.py with USE_CUDA=0 in the master branch, the test_operation.py cannot run before removing the gpu_dev related things.